### DISCUSSION TITLE
Restart button on GUI

### DISCUSSION THREAD

**Discussion by vicentecarro** [comfyanonymous/ComfyUI#1762] [Ideas]
Could you add a restart button in the GUI that allows users to restart comfyUI without having to access the terminal? In our use case we have ComfyUI in a local server. But every time someone installs something new, I need to connect to the server and restart the thing manually. That's a pain.
It would be great to have a button for that. 
Probably it would require to modify the launch script to wrap the actual launcher in an infinite loop or something, but nothing too hard I presume(at least not in Linux, no idea how to do that in Windows). 

Alternatively, maybe there is a way to hot-install new resources without having to restart the whole thing...

**Comment by ThisModernDay** [comment]
I went ahead and implemented a reboot server button and API route on the ComfyUI server and submitted it as a PR. However in the meantime here is a link to my branch with the reboot server button included.

https://github.com/ThisModernDay/ComfyUI/tree/reboot-button

![comfyui_screenshot.png](https://raw.githubusercontent.com/ThisModernDay/ComfyUI/reboot-button/rebootcomfyui.png)


Cheers!

**Reply by badgids** [reply]
All I had to do in my ComfyUI folder is:

```
git pull https://github.com/ThisModernDay/ComfyUI.git reboot-button
```

and I was good to go!

Thanks, @ThisModernDay !!!!!

**Reply by seghier** [reply]
I don't recommend this because will make a problem with portable ComfyUI

**Comment by vicentecarro** [comment]
Great. Thanks! I also think it would make more sense in a less visible place like the extra options.

**Reply by ThisModernDay** [reply]
https://github.com/comfyanonymous/ComfyUI/discussions/1762#discussioncomment-7352313

**Comment by ThisModernDay** [comment]
Updated Changes
-----------
- Added the ability to reboot the server with the click of a button
- Added an option in the settings menu to select the reboot server button location currently 3 locations are available. Menu Bar Top Icon, Small Button, Main Menu
- Added an option to toggle whether or not you have to confirm that you'd like to reboot the server

![Reboot Button Screenshot](https://github.com/ThisModernDay/ComfyUI/blob/reboot-button/rebootcomfyui.png?raw=true)

**Comment by badgids** [comment]
Ummm... It's already been implemented in the manager for several releases
now...

**Reply by ltdrdata** [reply]
The situation is slightly different. The restart button is currently implemented to be displayed only in limited scenarios, such as upon completion of tasks like an update or install. There's a possibility of it being added directly to ComfyUI, so its addition to the Manager Menu is being deferred.

I am waiting this PR.
https://github.com/comfyanonymous/ComfyUI/pull/1814

However, the issue is that an error occurs when there is an empty space in the path in the current implementation.

**Reply by seghier** [reply]
Portable version is not connected to Github and can't pull anything

**Reply by badgids** [reply]
Oh... We kind of abandoned the project when the restart button was implemented in the Manager, as all we were wanting it for was to reboot after a plugin re-install.

We can fix the path issue if you're wanting to pull it.

**Reply by ltdrdata** [reply]
I'd added 'restart' button to Manager menu.
You can submit PR to fix the path issue.
The original code is taken from that PR.

**Reply by seghier** [reply]
Thank you @ltdrdata , that's very useful

![image](https://github.com/comfyanonymous/ComfyUI/assets/6026588/c14f8a88-5617-45ab-a63e-35186e231dc2)
